src/sonic-config-engine/tests/common_utils.py

```diff
@@ -31,3 +37,47 @@ def liststr_to_dict(liststr):
 
     return list_obj
 
+class YangWrapper(object):
+    def __init__(self, path=YANG_MODELS_DIR):
+        """
+        sonic_yang only supports python3
+        """
+        if PY3x:
+            import sonic_yang
+            self.yang_parser = sonic_yang.SonicYang(path)
+            self.yang_parser.loadYangModel()
+            self.test_dir = os.path.dirname(os.path.realpath(__file__))
+            self.script_file = PYTHON_INTERPRETTER + ' ' + os.path.join(self.test_dir, '..', 'sonic-cfggen')
+
+    def validate(self, argument):
+        """
+        Raise exception when yang validation failed
+        """
+        if PY3x and "-m" in argument:
+            import sonic_yang
+            parser=argparse.ArgumentParser(description="Render configuration file from minigraph data and jinja2 template.")
+            parser.add_argument("-m", "--minigraph", help="minigraph xml file", nargs='?', const='/etc/sonic/minigraph.xml')
+            parser.add_argument("-k", "--hwsku", help="HwSKU")
+            parser.add_argument("-n", "--namespace", help="namespace name", nargs='?', const=None, default=None)
+            parser.add_argument("-p", "--port-config", help="port config file, used with -m or -k", nargs='?', const=None)
+            parser.add_argument("-S", "--hwsku-config", help="hwsku config file, used with -p and -m or -k", nargs='?', const=None)
+            args, unknown = parser.parse_known_args(shlex.split(argument))
+
+            print('\n    Validating yang schema')
+            cmd = self.script_file + ' -m ' + args.minigraph
+            if args.hwsku is not None:
+                cmd += ' -k ' + args.hwsku
+            if args.hwsku_config is not None:
+                cmd += ' -S ' + args.hwsku_config
+            if args.port_config is not None:
+                cmd += ' -p ' + args.port_config
+            if args.namespace is not None:
+                cmd += ' -n ' + args.namespace
+            cmd += ' --print-data'
+            output = subprocess.check_output(cmd, shell=True).decode()
+            try:
+                self.yang_parser.loadData(configdbJson=json.loads(output))
+                self.yang_parser.validate_data_tree()
+            except sonic_yang.SonicYangException as e:
+                print("yang data generated from %s is not valid"%(args.minigraph))
+                raise
```

Comment:
>raise [](http://example.com/codeflow?start=16&length=5)

`raise` is not friendly in unit test, You may return a bool, and a string. And the caller could use `assert` to apply the pair.

Validation failure should be treated as a test case failure, not an exception.